src/Infrastructure/Twig/AppExtension.php

```diff
@@ -39,22 +40,27 @@ public function formatDateTime(\DateTimeInterface $date, ?\DateTimeInterface $ti
         return $dateTime->format($format);
     }
 
-    public function isFuture(\DateTimeInterface $date = null, \DateTimeInterface $time = null, \DateTimeInterface $reference = null): bool
+    public function isClientPastDay(\DateTimeInterface $date, \DateTimeInterface $today = null): bool
     {
-        if (!$date) {
-            $date = new \DateTimeImmutable('now');
+        if (!$today) {
+            $today = (new \DateTimeImmutable('now'))->setTimeZone($this->clientTimezone)->setTime(0, 0, 0, 0);
+        } else {
+            $today = \DateTimeImmutable::createFromInterface($today)->setTimeZone($this->clientTimezone)->setTime(0, 0, 0, 0);
         }
 
-        if (!$reference) {
-            $reference = new \DateTimeImmutable('now');
-        }
+        $day = \DateTimeImmutable::createFromInterface($date)->setTimeZone($this->clientTimezone)->setTime(0, 0, 0, 0);
 
-        $dateTime = \DateTimeImmutable::createFromInterface($date);
+        return $day < $today;
+    }
 
-        if ($time) {
-            $dateTime = $dateTime->setTime((int) $time->format('H'), (int) $time->format('i'), (int) $time->format('s'));
+    public function isClientFutureDay(\DateTimeInterface $date, \DateTimeInterface $today = null): bool
+    {
+        if (!$today) {
+            $today = (new \DateTimeImmutable('now'))->setTimeZone($this->clientTimezone)->setTime(0, 0, 0, 0);
         }
```

Comment:
C'était une bonne idée dans l'autre fonction de convertir la timezone et heure de `$today` y compris s'il n'est pas `null`

Donc le bout de code serait le même

```suggestion
        $today = $today ? \DateTimeImmutable::createFromInterface($today) : new \DateTimeImmutable('now');
        $today = $today->setTimeZone($this->clientTimezone)->setTime(0, 0, 0, 0);
```